zap/src/output/typescript/server.rs

```diff
@@ -45,14 +46,27 @@ impl<'src> ServerOutput<'src> {
 	}
 
 	fn push_tydecl(&mut self, tydecl: &TyDecl) {
-		let name = &tydecl.name;
-		let ty = tydecl.ty.borrow();
-		let ty = &*ty;
+		let ty = &*tydecl.ty.borrow();
 
+		let mut depth = 0usize;
+		for name in &tydecl.path {
+			depth += 1;
+			self.push_indent();
+			self.push("export ");
+			if depth == 1 {
+				self.push("declare ");
+			}
+			self.push(&format!("namespace {name} {{\n"));
```

Comment:
Instead of doing namespaces here (aka creating one for each type), `push_tydecls` should just use `traverse_namespaces` so that we only declare two namespaces that are merged together